game/src/environment/towers/core/Tower.gd

```diff
@@ -28,6 +28,7 @@ export var range_radius = 528
 export var tower_damage = 20
 export var projectile_speed = 600
 export var tower_attack_speed = 1.0
+export var is_firing = false
```

Comment:
Am I right to say this step is to ensure towers carried by the mouse don't fire? I would recommend a different name; is_built instead, cuz is_firing implies a tower that is idling while waiting for an enmy to come would set is_firing to false.

Reply:
we can use it for both functionalities as well no?

Reply:
sorry idgi, i set out 2 behaviours:
1. is_firing is equivalent to whether tower is NOT being carried by mouse
2. is_firing is equivalent to whether or not it is firing (so a tower carried by mouse OR a tower idling waiting for an enemy both have is_firing = false)

I see from your code that the behaviour is actually number 1 and proposed a name change to is_built because is_firing would confuse ppl and imply number 2. But you're saying Number 2 is the actual way you intended is_firing to be used?

Reply:
clarified over tele, I understand now. 

tc: changing var name to is_frozen or smth else instead.